tests/lightgbm/test_lightgbm_autolog.py

```diff
@@ -0,0 +1,270 @@
+import os
+import json
+import pytest
+import numpy as np
+import pandas as pd
+from sklearn import datasets
+import lightgbm as lgb
+
+import mlflow
+import mlflow.lightgbm
+
+client = mlflow.tracking.MlflowClient()
+
+
+def get_latest_run():
+    return client.get_run(client.list_run_infos(experiment_id='0')[0].run_id)
+
+
+@pytest.fixture(scope="session")
+def bst_params():
+    return {
+        'objective': 'multiclass',
+        'num_class': 3,
+    }
+
+
+@pytest.fixture(scope="session")
+def train_set():
+    iris = datasets.load_iris()
+    X = pd.DataFrame(iris.data[:, :2], columns=iris.feature_names[:2])
+    y = iris.target
+    # set free_raw_data False to use raw data later.
+    return lgb.Dataset(X, y, free_raw_data=False)
+
+
+@pytest.mark.large
+def test_lgb_autolog_ends_auto_created_run(bst_params, train_set):
+    mlflow.lightgbm.autolog()
+    lgb.train(bst_params, train_set)
+    assert mlflow.active_run() is None
+
+
+@pytest.mark.large
+def test_lgb_autolog_persists_manually_created_run(bst_params, train_set):
+    mlflow.lightgbm.autolog()
+    with mlflow.start_run() as run:
+        lgb.train(bst_params, train_set)
+        assert mlflow.active_run()
+        assert mlflow.active_run().info.run_id == run.info.run_id
+
+
+@pytest.mark.large
+def test_lgb_autolog_logs_default_params(bst_params, train_set):
+    mlflow.lightgbm.autolog()
+    lgb.train(bst_params, train_set)
+    run = get_latest_run()
+    params = run.data.params
+
+    expected_params = {
+        'num_boost_round': 100,
+        'feature_name': 'auto',
+        'categorical_feature': 'auto',
+        'verbose_eval': True,
+        'keep_training_booster': False
+    }
+    expected_params.update(bst_params)
+
+    for key, val in expected_params.items():
+        assert key in params
+        assert params[key] == str(val)
+
+    unlogged_params = ['params', 'train_set', 'valid_sets', 'valid_names', 'fobj', 'feval',
+                       'init_model', 'evals_result', 'learning_rates', 'callbacks']
+
+    for param in unlogged_params:
+        assert param not in params
+
+
+@pytest.mark.large
+def test_lgb_autolog_logs_specified_params(bst_params, train_set):
+    mlflow.lightgbm.autolog()
+    expected_params = {
+        'num_boost_round': 10,
```

Comment:
How long do these tests take to run btw? Could we speed them up by training for fewer epochs, or is the duration negligible (<5 seconds or so)? Mainly just wondering if we can save some CI time

Reply:
I measured how it takes to run the training part (`lgb.train(...)`). It took `0.667` seconds on my macbook pro (16GB RAM).

Reply:
However, some tests use the default `num_boost_round` value (100) which I think is too large. I will fix these tests to use fewer epochs.